### DISCUSSION TITLE
usePagination hook with 6.28 in React 18 (CRA) routing fails

### DISCUSSION THREAD

**Discussion by ronaldh46** [algolia/instantsearch#5323] [Q&A] [answered]
I have a React 18 project instantiated using Create React App. I'm using the latest 6.28 build of `react-instantsearch-hooks-web`.

The problem:
- When using the `usePagination` hook, and routing, the `refinePage` and/or the `createURL` functions attempt to update the the routing/URL, but appears to cause a re-render, preventing navigation between pages of search results.
- There is a flash of the target page, before the initial page loads.
- It appears the pagination query parameter is stripped from the URL.
- If I apply the page parameters directly in the URL, it the above steps are also reproducible.
- Interestingly, using `useRefinement` works, despite the state being stored as query params (default configuration).
- When I remove the other hooks, `usePagination` works.

Is this an issue with Strict Mode or something I can address? I am using the default routing config.

**Comment by francoischalifour** [comment]
Can you please provide a reproducible sandbox on [CodeSandbox](https://codesandbox.io/)?

**Comment by VegtMedia-Alex** [comment]
The dispose method of the router is triggered right after page load. Which causes the reload and afterwards it will fail because it can not find the mainIndex. I think this is related to this issue:
https://github.com/algolia/react-instantsearch/issues/3530

**Reply by ronaldh46** [reply]
Not sure.

I have fixed the paging issue (so navigation between pages of results) once a hits have been loaded work, but rehydrating results from the search params in the routing still doesn't work.
(the new `useInstantSearch` hook was causing a re-render, I was using this to detect if there was an empty query; now using alternative method)

Looking into it further, I can navigate/refine search results, and URL updates correctly:
E.g.
`http://localhost:3000/?search_version2%5Bquery%5D=a&search_version2%5BrefinementList%5D%5Bcategory.lvl2%5D%5B0%5D=Devices`

If I copy and paste and use that URL directly, the browser redirects to this:
`http://localhost:3000/?search_version2%25255Bquery%25255D=a&search_version2%25255BrefinementList%25255D%25255Bcategory.lvl2%25255D%25255B0%25255D=Devices`

Then clears all params to this
`http://localhost:3000/`

I suspect it might be a routing/history issue.

**Comment by francoischalifour** [comment, marked as answer]
This issue could be fixed in the [v6.30.2](https://github.com/algolia/react-instantsearch/releases/tag/v6.30.2) release. Please upgrade and let us know how it goes!
